packages/store/tests/state-token.spec.ts

```diff
@@ -15,45 +15,51 @@ describe('[TEST]: StateToken', () => {
   });
 
   describe('Integration', () => {
-    it('should successfully create store', () => {
-      const TODO_LIST_TOKEN = new StateToken<string[]>('todoList');
+    const TODO_LIST_TOKEN = new StateToken<string[]>('todoList');
 
-      @State<string[]>({
-        name: TODO_LIST_TOKEN,
-        defaults: ['hello', 'world']
-      })
-      @Injectable()
-      class TodoListState {
-        @Selector([TODO_LIST_TOKEN])
-        static todos(state: string[]) {
-          return state;
-        }
+    @State<string[]>({
+      name: TODO_LIST_TOKEN,
+      defaults: ['hello', 'world']
+    })
+    @Injectable()
+    class TodoListState {
+      @Selector([TODO_LIST_TOKEN])
+      static todos(state: string[]) {
+        return state;
+      }
 
-        @Selector([TODO_LIST_TOKEN])
-        static first(state: string[]): string {
-          return state.slice().shift()!;
-        }
+      @Selector([TODO_LIST_TOKEN])
+      static first(state: string[]): string {
+        return state.slice().shift()!;
       }
+    }
 
-      @Component({
-        selector: 'myApp',
-        template: '{{ myState$ | async | json }}'
-      })
-      class MyComponent {
-        @Select(TODO_LIST_TOKEN)
-        myState$: Observable<string[]>;
+    @Component({
+      selector: 'myApp',
+      template: '{{ myState$ | async | json }}'
+    })
+    class MyComponent {
+      @Select(TODO_LIST_TOKEN)
+      myState$: Observable<string[]>;
 
-        constructor(public storeApp: Store) {}
-      }
+      constructor(public storeApp: Store) {}
+    }
 
-      TestBed.configureTestingModule({
+    let fixture: ComponentFixture<MyComponent>;
+
+    beforeEach(async () => {
+      await TestBed.configureTestingModule({
         imports: [NgxsModule.forRoot([TodoListState])],
         declarations: [MyComponent]
-      });
-
-      const fixture = TestBed.createComponent(MyComponent);
-      fixture.autoDetectChanges();
+      })
+        .compileComponents()
+        .then(() => {
+          fixture = TestBed.createComponent(MyComponent);
+          fixture.autoDetectChanges();
+        });
+    });
 
```

Comment:
It is preferable for this to either be inside the unit test, or moved out to a `setupFixture` function that returns the specific references that are needed. Was there a specific reason for moving this out of the `it` and using the `beforeEach`?

Reply:
@markwhitfeld Of course it can be a matter of preference, there are different ways to do things.
Below are my reasons for making this change.
Each `describe` function is a kind of broader context for several test cases that are wrapped in the `it` functions. In my experience, it's best to remove as many initializations and declarations from the `it` functions as possible to keep the `it` functions concise and easy to read.
Also, the TestBed is initialized and components are compiled asynchronously before each `it` funciton executes. It is a standard and well known pattern to write unit tests as far as I know.
Actually, it is the way the Angular CLI generates empty unit specs. My experience has shown that this approach requires less maintenance costs for large projects.
As for the function that returns needed variables after, I have never used such approach in unit tests. Mainly I use block scoping.

So the main reason is readability and ease of maintenance.

Not that I insist on any specific implementation, you are the maintainer of the project, feel free to make any changes to the PR as necessary.

Reply:
@markwhitfeld updated, please check if this works https://github.com/ngxs/store/pull/1846/commits/e09d6c69e86fdace0ac031ee97b48cc32e35d821

Reply:
Thank you for making this update.
Hopefully it also appears as a cleaner approach to you too 😉